crates/iota-indexer/src/indexer_reader.rs

```diff
@@ -631,19 +631,43 @@ impl IndexerReader {
         let digests = digests
             .iter()
             .map(|digest| digest.inner().to_vec())
-            .collect::<Vec<_>>();
-        run_query!(&self.pool, |conn| {
+            .collect::<HashSet<_>>();
+        let checkpointed_txs = run_query!(&self.pool, |conn| {
             transactions::table
                 .inner_join(
                     tx_digests::table
                         .on(transactions::tx_sequence_number.eq(tx_digests::tx_sequence_number)),
                 )
                 // we filter the tx_digests table because it is indexed by digest,
                 // transactions table is not
-                .filter(tx_digests::tx_digest.eq_any(digests))
+                .filter(tx_digests::tx_digest.eq_any(&digests))
                 .select(StoredTransaction::as_select())
                 .load::<StoredTransaction>(conn)
-        })
+        })?;
+        if checkpointed_txs.len() == digests.len() {
+            return Ok(checkpointed_txs);
+        }
+        let loaded_digests = checkpointed_txs
+            .iter()
+            .map(|tx| tx.transaction_digest.clone())
+            .collect();
+        let missing_digests = digests.difference(&loaded_digests);
```

Comment:
Consider this approach instead:
```rust
        for tx in &checkpointed_txs {
            digests.remove(&tx.transaction_digest)
        }
```
The remaining `digests` are the missing one. This is simpler and should be more efficient.

Reply:
Coupled it with variable rename, but otherwise the performance should be the same I suppose https://github.com/iotaledger/iota/pull/6242/commits/9b84dcae05b835a2c60a312107931a5af368b3b9